src/fileutil.c

```diff
@@ -15,29 +15,94 @@
  */
 
 #include <defines.h>
+#include <errno.h>
+#include <pwd.h>                /* for getpwnam_r() */
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 
 extern char* scenario_path;
 
+int expand_user_path(const char* path, char* expanded_home_path /*The buffer*/, size_t buflen)
+{
+    if (path[0] != '~') {                       /* We have nothing to expand here */
+        return 1;
+    }
+
+    memset(expanded_home_path, '\0', buflen);
+    char* home_dir = NULL;
+
+    if (path[1] == '\0' || path[1] == '/') {                                                    /* '~/path' case */
+        home_dir = getenv("HOME");
+        if (home_dir == NULL) {
+            home_dir = getenv("USERPROFILE");
+        } else {
+            snprintf(expanded_home_path, buflen - 1, "%s%s", home_dir, path + 1);
+        }
+    } else {
+        const char* first_slash = NULL;
+        first_slash = strchr(path, '/');                                                        /* substring starting from '/' */
```

Comment:
I googled a bit but couldn't find any reason not to, could you please tell me if there is any disadvantage of explicitly initializing it to `NULL` , so that I won't forget it in future.

Reply:
But if you assign it directly, it *is* initialized. I meant: 
```c
const char* first_slash = strchr(path, '/');                                                        /* substring starting from '/' */
```

Reply:
Sorry for not catching it earlier, updated now. Thanks!